torch/_inductor/kernel/mm_scaled_grouped.py

```diff
@@ -0,0 +1,434 @@
+
+        M_PER_GROUP = M // G
+        MIN_M_TILES = 32 if torch.version.hip else 64
+        # 2. make sure we don't load M tiles that are too big
+        if (
+            not use_warp_specialization
+            and BLOCK_M > MIN_M_TILES
+            and BLOCK_M > (M_PER_GROUP * 2)
+        ):
+            continue
+        # 3. make sure we don't load N tiles that are too small
+        if BLOCK_M < 128 and BLOCK_M < (M_PER_GROUP // 2):
+            continue
+
+        num_sm = driver.active.utils.get_device_properties(device)[
+            "multiprocessor_count"
+        ]
+        N_TILES = N // BLOCK_N
+        MIN_N_TILES = 32 if torch.version.hip else 64
+        # 4. make sure we don't load N tiles that are too big
+        if (
+            not use_warp_specialization
+            and BLOCK_N > MIN_N_TILES
+            and M * N_TILES < num_sm
+        ):
+            continue
+        # 5. make sure we don't load N tiles that are too small
+        if BLOCK_N < 128 and M * N_TILES > 2 * num_sm:
+            continue
+
+        # 6. make sure K can be evenly divided
+        if K % BLOCK_K != 0:
+            continue
+
+        # 7. make sure we can partition for ws
+        if use_warp_specialization:
+            if num_warps != 4:
+                continue
+
+            # "tritongpu-warp-spec-data-partition"
+            m_slice = BLOCK_M // num_consumer_groups
+            n_slice = BLOCK_N // num_consumer_groups
+            if m_slice < 64 and n_slice < 256:
+                continue
+
+        pruned_configs.append(config)
+
+    return pruned_configs
+
+
+# Copied from fbgemm grouped_gemm.py
+triton_scaled_grouped_mm_source = r"""
+{{def_kernel("a_ptr", "b_ptr", "a_scale_ptr", "b_scale_ptr", "m_sizes")}}
+    tidx = tl.program_id(0)
+
+    dtype = tl.float8e4nv
+    TMA_SIZE: tl.constexpr = tl.constexpr(128)
+
+    workspace_base = ws_ptr + tidx * 2 * TMA_SIZE
+    c_desc_ptr = None
+
+    a_desc_ptr = workspace_base
+    b_desc_ptr = workspace_base + TMA_SIZE
+
+    triton.language.extra.cuda.experimental_device_tensormap_create2d(
+        desc_ptr=a_desc_ptr,
+        global_address=a_ptr,
+        load_size=[BLOCK_M, BLOCK_K],
+        global_size=[M, K],
+        element_ty=a_ptr.dtype.element_ty,
+    )
+    triton.language.extra.cuda.experimental_device_tensormap_create2d(
+        desc_ptr=b_desc_ptr,
+        global_address=b_ptr,
+        load_size=[BLOCK_N, BLOCK_K],
+        global_size=[N * G, K],
+        element_ty=b_ptr.dtype.element_ty,
+    )
+    tl.extra.cuda.experimental_tensormap_fenceproxy_acquire(a_desc_ptr)
+    tl.extra.cuda.experimental_tensormap_fenceproxy_acquire(b_desc_ptr)
+
+    M_end_offset = 0
+    iterated_tiles = 0
+    for g in tl.range(G):
+        # Move across groups
+        M_start_offset = M_end_offset
+        M_end_offset = tl.load(m_sizes + g)
+        m_size = M_end_offset - M_start_offset
+
+        if m_size > 0:
+            N_start_offset = g.to(tl.int64) * N
```

Comment:
do we want to gate tl.int64 on input sizes ?

Reply:
I think this is a good idea but maybe want to save it for an optimization down the line; I don't think it significantly slows down this kernel as-is